admin/menu/class-base-menu.php

```diff
@@ -0,0 +1,265 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin\Menu
+ */
+
+/**
+ * Admin menu base class.
+ */
+abstract class WPSEO_Base_Menu implements WPSEO_WordPress_Integration {
+
+	/** @var WPSEO_Menu Menu */
+	protected $menu;
+
+	/**
+	 * Constructs the Admin Menu.
+	 *
+	 * @param WPSEO_Menu $menu Menu to use.
+	 */
+	public function __construct( WPSEO_Menu $menu ) {
+		$this->menu = $menu;
+	}
+
+	/**
+	 * Registers all hooks to WordPress.
+	 *
+	 * @return void
+	 */
+	abstract public function register_hooks();
+
+	/**
+	 * Returns the list of registered submenu pages.
+	 *
+	 * @return array List of registered submenu pages.
+	 */
+	abstract public function get_submenu_pages();
+
+	/**
+	 * Creates a submenu formatted array.
+	 *
+	 * @param string     $page_title Page title to use.
+	 * @param string     $page_slug  Page slug to use.
+	 * @param callable   $callback   Optional. Callback which handles the page request.
+	 * @param callable[] $hook       Optional. Hook to trigger when the page is registered.
```

Comment:
I'm unsure whether we use the `[]` in our documentation. I'd suggest removing it.